### DISCUSSION TITLE
Fetch latest schema with CLI

### DISCUSSION THREAD

**Discussion by PlayAnyData** [graphql-hive/console#5626] [Help] [answered]
As far as I can see, there is currently no option to fetch the latest schema via HIVE CLI, although, there is an option in the GQL API:
https://github.com/kamilkisiela/graphql-hive/blob/498f621e7dfda778c02130b22e3f1ccf9e303af9/packages/services/api/src/modules/schema/module.graphql.ts#L59C5-L59C18

The CLI only offers to fetch by action id. However, this is usually not known to our clients. Would be great to allow a fetch by "latest".

**Comment by n1ru4l** [comment, marked as answer]
Hey, you can fetch the latest schema SDL/Supergraph from the CDN via the `artifact:fetch` command: https://github.com/kamilkisiela/graphql-hive/blob/main/packages/libraries/cli/README.md#hive-artifactfetch

**Reply by PlayAnyData** [reply]
First of all: Thanks a lot for the answer.

However: I should have written: We are self-hosting hive. I have tried the following to fetch the artifact:
* Connecting to server w and without graphql suffix
* Connecting to schema pod
* Connecting to s3proxy

Nothing, seems to work, as I am getting 404 for all of them. Could you point me in the right direction, please?

**Reply by dotansimha** [reply]
You should point your CLI to your self-hosted Hive instance, check: https://the-guild.dev/graphql/hive/docs/self-hosting/client-and-cli-configuration

**Reply by PlayAnyData** [reply]
@dotansimha 
I am sorry, but this does not work: First of all it seems cdn.endpoint is (contrasting to documentation) required. And second, independent of which type of artifact I select, I get the following issue:
```
C:\Users\bla\IdeaProjects\dsl> npx hive artifact:fetch --artifact services --cdn.endpoint "https://graphql-hive-server-dataservicelayer-dev-dsl.devdsl.traefik.eu.corporation.corp/graphql" --cdn.accessToken "token" --debug
GET https://graphql-hive-server-dataservicelayer-dev-dsl.devdsl.traefik.eu.techem.corp/graphql/services Attempt (1/4)
GET https://graphql-hive-server-dataservicelayer-dev-dsl.devdsl.traefik.eu.techem.corp/graphql/services failed with status 404 (385ms): {"message":"Route GET:/graphql/services not found","error":"Not Found","statusCode":404}
Abort retry because of status code 404.
    Error: GET https://graphql-hive-server-dataservicelayer-dev-dsl.devdsl.traefik.eu.techem.corp/graphql/services failed with status 404.
```

**Reply by PlayAnyData** [reply]
I am using latest version of cli.

**Reply by n1ru4l** [reply]
@PlayAnyData You need to provide the CDN endpoint and CDN access token, not the GraphQL endpoint and registry token.

You can find the CDN URL here: 
![image](https://github.com/user-attachments/assets/3610645f-b6e0-4414-bbaa-9d5a5399a966)


**Example:**

```bash
hive artifact:fetch \
  --cdn.endpoint "https://xxxxxxxxx.com/artifacts/v1/5d80a1c2-2532-419c-8bb5-75bb04ea1112" \
  --cdn.accessToken "hv2...." \
  --artifact sdl
```

**Reply by PlayAnyData** [reply]
@n1ru4l Ah, okay. Gotcha, thanks a lot!

I have found this now for self-hosting: https://the-guild.dev/graphql/hive/docs/self-hosting/cdn-artifacts
